server/protocols/webadmin/webadmin-jmap/src/main/java/org/apache/james/webadmin/data/jmap/MessageFastViewProjectionCorrector.java

```diff
@@ -53,6 +55,74 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(MessageFastViewProjectionCorrector.class);
 
+    public static class RunningOptions {
+        public static class Builder {
+            private Optional<Integer> messageParallelism;
+            private Optional<Integer> mailboxParallelism;
+            private Optional<Integer> userParallelism;
+
+            public Builder() {
+                messageParallelism = Optional.empty();
+                mailboxParallelism = Optional.empty();
+                userParallelism = Optional.empty();
+            }
+
+            public Builder messageParallelism(int messageParallelism) {
+                this.messageParallelism = Optional.of(messageParallelism);
+                return this;
+            }
+
+            public Builder mailboxParallelism(int mailboxParallelism) {
+                this.mailboxParallelism = Optional.of(mailboxParallelism);
+                return this;
+            }
+
+            public Builder userParallelism(int userParallelism) {
+                this.userParallelism = Optional.of(userParallelism);
+                return this;
+            }
+
+            public RunningOptions build() {
+                return new RunningOptions(
+                    messageParallelism.orElse(1),
+                    mailboxParallelism.orElse(1),
+                    userParallelism.orElse(1));
+            }
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        public static RunningOptions DEFAULT = RunningOptions.builder().build();
+
+        private final int messageParallelism;
+        private final int mailboxParallelism;
+        private final int userParallelism;
+
+        public RunningOptions(int messageParallelism, int mailboxParallelism, int userParallelism) {
+            Preconditions.checkArgument(messageParallelism > 0, "'messageParallelism' must be stictly positive");
+            Preconditions.checkArgument(mailboxParallelism > 0, "'mailboxParallelism' must be stictly positive");
+            Preconditions.checkArgument(userParallelism > 0, "'userParallelism' must be stictly positive");
+
+            this.messageParallelism = messageParallelism;
+            this.mailboxParallelism = mailboxParallelism;
+            this.userParallelism = userParallelism;
+        }
+
+        public int getMessageParallelism() {
```

Comment:
can we use concurrency instead of parallelism?

Reply:
can you explain me why concurrency is better than parallelism in this context? I don't get it

Reply:
parallelism is a way to achieve concurrency. It means: "I will run those two computations at the same time on two different CPUs" for example.
If you do IO, you don't do parallel things, you do interleaved things.
The rule of thumb is: always use `concurrent` unless it's really important to state how you do concurrent things and they happen to be parallel.